From 83b914214ad2aaca7f4acc2b442a4e36d7f9c7d3 Mon Sep 17 00:00:00 2001
From: me-cafebabe <me.cafebabe@gmail.com>
Date: Thu, 5 Oct 2023 19:38:27 +0000
Subject: [PATCH 2/2] libc: libstdc++: Rename vendor variant to
 libstdc++_vendor

* In Android 14, This library got listed on /vendor/etc/linker.config.pb,
  which is duplicated with /system/etc/linker.config.pb, as a result
  linkerconfig fails and stops the entire system from booting.

[    5.840677] DEBUG: Abort message: 'duplicate: libstdc++.so is provided by default and system in [vendor]'

Change-Id: Ib6eee48de87100c2fb0dd765d57ee95d3159c4ef
---
 libc/Android.bp | 16 +++++++++++++---
 1 file changed, 13 insertions(+), 3 deletions(-)

diff --git a/libc/Android.bp b/libc/Android.bp
index f652a33db..71d8ee556 100644
--- a/libc/Android.bp
+++ b/libc/Android.bp
@@ -2084,7 +2084,7 @@ cc_library_headers {
 // libstdc++.so and libstdc++.a.
 // ========================================================
 
-cc_library {
+cc_defaults {
     defaults: ["libc_defaults"],
     include_dirs: ["bionic/libstdc++/include"],
     srcs: [
@@ -2092,13 +2092,12 @@ cc_library {
         "bionic/__cxa_pure_virtual.cpp",
         "bionic/new.cpp",
     ],
-    name: "libstdc++",
+    name: "libstdc++_defaults",
     static_ndk_lib: true,
     static_libs: ["libasync_safe"],
     apex_available: [
         "//apex_available:platform",
     ],
-    vendor_available: true,
 
     static: {
         system_shared_libs: [],
@@ -2136,6 +2135,17 @@ cc_library {
     },
 }
 
+cc_library {
+    name: "libstdc++",
+    defaults: ["libstdc++_defaults"],
+}
+
+cc_library_shared {
+    name: "libstdc++_vendor",
+    defaults: ["libstdc++_defaults"],
+    vendor: true,
+}
+
 genrule {
     name: "libstdc++.arm.map",
     out: ["libstdc++.arm.map.txt"],
-- 
2.39.2

